clang/lib/CConv/ConstraintVariables.cpp

```diff
@@ -1629,16 +1631,12 @@ void FunctionVariableConstraint::brainTransplant(ConstraintVariable *FromCV,
   auto RetVar = getOnly(returnVars);
   RetVar->brainTransplant(FromRetVar, I);
   // Transplant params.
-  assert(From->numParams() == numParams());
-  for (unsigned i = 0; i < From->numParams(); i++) {
-    CVarSet &FromP = From->getParamVar(i);
-    CVarSet &P = getParamVar(i);
-    auto FromVar = getOnly(FromP);
-    auto Var = getOnly(P);
-    Var->brainTransplant(FromVar, I);
-  }
+  handle_params(From, this,
+                [&I](ConstraintVariable *F, ConstraintVariable *T)
+                { T->brainTransplant(F, I); });
 }
 
+
 void FunctionVariableConstraint::mergeDeclaration(ConstraintVariable *FromCV) {
```

Comment:
I'm fuzzy -- when do we use `brainTransplant` vs. `mergeDeclaration` ? I don't know how that affects what might be needed in the two cases.

Reply:
Looks like `mergeDeclaration` is used when when combining declarations, and `brainTransplant` is used when combining definitions. Changing the interface for `mergeDeclaration` would get rid of the need to track the ProgramInfo references though, since both are called from inside ProgramInfo itself.

Reply:
Again, don't follow.

I am skeptical about your characterization of `brainTransplant` vs. `mergeDeclaration`. We can't combine definitions. There is only one definition and potentially many declarations.

Reply:
Was reading from ProgramInfo:483. It's was my understanding the mergeDelcartion is used for merging two declarations, whereas braintransplant is used for merging a definition with a declaration.

Reply:
Yes, merging a decl with a def sounds right for brainTransplant, where it's in that order IIRC. If you see the def first, I don't think we do brainTransplant, do we?